crates/matrix-sdk/tests/integration/event_cache.rs

```diff
@@ -136,6 +138,117 @@ async fn test_add_initial_events() {
     assert!(subscriber.is_empty());
 }
 
+#[async_test]
+async fn test_ignored_unignored() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    // Immediately subscribe the event cache to sync updates.
+    client.event_cache().subscribe().unwrap();
+
+    // If I sync and get informed I've joined The Room, but with no events,
+    let room_id = room_id!("!omelette:fromage.fr");
+    let other_room_id = room_id!("!galette:saucisse.bzh");
+
+    let mut sync_builder = SyncResponseBuilder::new();
+    sync_builder
+        .add_joined_room(JoinedRoomBuilder::new(room_id))
+        .add_joined_room(JoinedRoomBuilder::new(other_room_id));
+
+    mock_sync(&server, sync_builder.build_json_sync_response(), None).await;
+    client.sync_once(Default::default()).await.unwrap();
+    server.reset().await;
+
+    let dexter = user_id!("@dexter:lab.org");
+    let ivan = user_id!("@ivan:lab.ch");
```

Comment:
😉

Reply:
could be any Ivan in Switzerland :see_no_evil:

Reply:
Suuuuuuure